src/game/screen.c

```diff
@@ -96,6 +106,21 @@ int32_t Screen_GetRenderScale(int32_t unit)
     return MIN(scale_x, scale_y);
 }
 
+int32_t Screen_GetRenderScaleBar(int32_t unit)
+{
+    int32_t baseWidth = 640;
+    int32_t baseHeight = 480;
+    int32_t scale_x = Screen_GetResWidth() > baseWidth
+        ? ((double)Screen_GetResWidth() * unit * g_Config.ui.bar_scale)
+            / baseWidth
+        : unit * g_Config.ui.bar_scale;
+    int32_t scale_y = Screen_GetResHeight() > baseHeight
+        ? ((double)Screen_GetResHeight() * unit * g_Config.ui.bar_scale)
+            / baseHeight
+        : unit * g_Config.ui.bar_scale;
+    return MIN(scale_x, scale_y);
+}
```

Comment:
These two functions are practically identical. I suggest to refactor things a bit:

```c
int32_t Screen_GetRenderScaleBase(
    int32_t unit, int32_t base_width, int32_t base_height, double factor
) {
￼    int32_t scale_x = Screen_GetResWidth() > base_width
￼        ? ((double)Screen_GetResWidth() * unit * factor)
￼            / base_width
￼        : unit * factor;
￼    int32_t scale_y = Screen_GetResHeight() > base_height
￼        ? ((double)Screen_GetResHeight() * unit * factor)
￼            / base_height
￼        : unit * factor;
￼    return MIN(scale_x, scale_y);
}

int32_t Screen_GetRenderScaleText(int32_t unit) {
    return Screen_GetRenderScaleBase(unit, 640, 480, g_Config.ui.text_scale);
}

int32_t Screen_GetRenderScaleBar(int32_t unit) {
    return Screen_GetRenderScaleBase(unit, 640, 480, g_Config.ui.bar_scale);
}
```

and replace `Screen_GetRenderScale` usages with `Screen_GetRenderScaleText`.